### PR TITLE
feat: cowswap executeTrade and getTradeTxs methods

### PR DESCRIPTION
**Pull request by chxash**
This implements both methods for `CowSwapper`.
This has been tested in web to successfully perform a swap with native wallet, metamask and metamask + ledger

Steps to test : 

- Build this lib branch
- Use following web branch with local lib : https://github.com/shapeshift/web/tree/cowswapper-web-integration-wip
- Perform any ERC20 swap (no ETH) with native / metamask

Swap with native wallet : 

native1

native2

Linked TX : https://etherscan.io/tx/0xf7d251e527c19db3ef68063f99c5cd98cb4d86959382e8043303743d23cfb9cf

Swap with metamask : 

trade_cow_product

Linked TX : https://etherscan.io/tx/0xcec287df80d98ce1d42b26f97260d9aea89e7daa551f388622bb4ca02978a049

### CODE REVIEW

**Review by gomesalexandre — COMMENTED** [review]
A few early comments, will give another pass to this after https://github.com/shapeshift/hdwallet/pull/546 is published

File: packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.ts [hunk 1]
```diff
@@ -86,50 +86,41 @@ export async function cowBuildTrade(
...
-    // feeInFeeAsset * feeAssetUsdRate = feeInSellAsset * sellAssetUsdRate
-    // hence feeInFeeAsset = feeInSellAsset * sellAssetUsdRate / feeAssetUsdRate
-    const feeInSellAsset = bnOrZero(quote.feeAmount).div(
-      bn(10).exponentiatedBy(sellAsset.precision)
-    )
-
-    // feeInFeeAsset = feeInSellAsset * sellAssetUsdRate / feeAssetUsdRate
-    const feeInFeeAsset = feeInSellAsset
+    // calculating trade fee in USD
+    const tradeFeeInUsd = bnOrZero(quote.feeAmount)
```
[the remark this answers is not in the document]
Comment (chxash) [reply packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.ts]: Done !

File: packages/swapper/src/swappers/cow/cowGetTradeTxs/cowGetTradeTxs.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import { SwapError, SwapErrorTypes, TradeResult, TradeTxs } from '../../../api'
+import { CowSwapperDeps } from '../CowSwapper'
+import { CowSwapGetOrdersResponse, CowSwapGetTradesResponse } from '../types'
+import { ORDER_STATUS_FULFILLED } from '../utils/constants'
+import { cowService } from '../utils/cowService'
+
+export async function cowGetTradeTxs(deps: CowSwapperDeps, input: TradeResult): Promise<TradeTxs> {
+  try {
+    const getOrdersResponse: AxiosResponse<CowSwapGetOrdersResponse> =
```
[the remark this answers is not in the document]
Comment (chxash) [reply packages/swapper/src/swappers/cow/cowGetTradeTxs/cowGetTradeTxs.ts]: done x2

File: packages/swapper/src/swappers/cow/utils/helpers/helpers.ts [hunk 3]
```diff
@@ -11,6 +13,41 @@ import { cowService } from '../cowService'
...
+  { name: 'kind', type: 'string' },
+  { name: 'partiallyFillable', type: 'bool' },
+  { name: 'sellTokenBalance', type: 'string' },
+  { name: 'buyTokenBalance', type: 'string' }
+]
+
+/**
+ * EIP-712 typed data type definitions.
+ */
+export declare type TypedDataTypes = Parameters<typeof ethers.utils._TypedDataEncoder.hashStruct>[1]
```
[the remark this answers is not in the document]
Comment (chxash) [reply packages/swapper/src/swappers/cow/utils/helpers/helpers.ts]: Oops yes ! done

File: packages/swapper/src/swappers/cow/utils/helpers/helpers.ts [hunk 4]
```diff
@@ -79,3 +116,35 @@ export const getNowPlusThirtyMinutesTimestamp = (): number => {
   ts.setMinutes(ts.getMinutes() + 30)
   return Math.round(ts.getTime() / 1000)
 }
+
+export const hashTypedData = (
+  domain: TypedDataDomain,
+  types: TypedDataTypes,
+  data: Record<string, unknown>
```
[the remark this answers is not in the document]
Comment (chxash) [reply packages/swapper/src/swappers/cow/utils/helpers/helpers.ts:123]: I dont see how we can do better yes :(

**Review by gomesalexandre — APPROVED** [review]
Tested out locally with MM and managed to succesfully trade ERC-20s using the CowSwap swapper 🎉 . Tx is properly populated in Transaction History components on your web branch. 

Note that on this branch, accounts without enough ETH for ZRX swaps will error because of `hasEnoughBalanceForGas` being false, and I had to use an account with enough ETH for fees, despite Cowswap Txs being gasless.

Irrelevant since the web PR is just for demo purposes, but when making the web PR for this, we'll have to make sure not to get regular ZRX fees estimation / not to handle them in `<TradeInput />` in the case of moo swaps

🥛

**Comment by chxash** [comment]
> Note that on this branch, accounts without enough ETH for ZRX swaps will error because of `hasEnoughBalanceForGas` being false, and I had to use an account with enough ETH for fees, despite Cowswap Txs being gasless.

@gomesalexandre interesting finding, thank you !

**Review by 0xdef1cafe — APPROVED** [review]
after we have cowswap, osmo, thorchain, zrx across eth and avalanche we can revisit and tidy abstractions, but this is good for now

**Review by gomesalexandre — APPROVED** [review]
LGTM

Tested again with your latest changes and I was able to make a moo swap with your latest changes and I was able to get cowswaps working both with an account that has enough ETH for gas, and an account that doesn't have enough ETH for regular swaps

🐮

**Pull request merged**